### PR TITLE
Message bus tests

### PR DESCRIPTION
**Pull request by piotr921**
Add tests for message bus, refactor structure of this class.

### CODE REVIEW

File: src/main/java/com/spanish_inquisition/battleship/server/MessageBus.java [hunk 1]
```diff
@@ -32,36 +31,37 @@ Message getMessageFor(int recipientId) {
...
 
-    boolean haveMessageFromSender(int senderId) {
-        if (messageBus != null) {
-            for (Message message : messageBus)
-                if (message.getSender() == senderId) return true;
+    Message getMessageFrom(int senderId) {
+        Message firstSenderMessage;
+        try {
+            firstSenderMessage = messageBus.stream()
+                    .filter(message -> message.isFromSender(senderId)).findFirst().get();
```
Comment (mprtcz) [inline src/main/java/com/spanish_inquisition/battleship/server/MessageBus.java]: Dat labda ❤️

File: src/main/java/com/spanish_inquisition/battleship/server/MessageBus.java [hunk 2]
```diff
@@ -32,36 +31,37 @@ Message getMessageFor(int recipientId) {
...
     }
 
-    boolean haveMessageForRecipient(int recipientId) {
-        if (messageBus != null) {
-            for (Message message : messageBus)
-                if (message.isToRecipient(recipientId)) return true;
-        }
-        return false;
+    private void removeAllSenderMessagesFromBus(int senderId) {
+        messageBus.removeIf(message -> message.isFromSender(senderId));
```
Comment (mprtcz) [inline src/main/java/com/spanish_inquisition/battleship/server/MessageBus.java]: Lambada ❤️

File: src/main/java/com/spanish_inquisition/battleship/server/MessageBus.java [hunk 3]
```diff
@@ -32,36 +31,37 @@ Message getMessageFor(int recipientId) {
...
-        List<Message> newList = new LinkedList<>();
-        for (Message msg : messageBus) {
-            if (msg.getSender() == senderId) {
-                newList.add(msg);
-            }
+    boolean haveMessageFromSender(int senderId) {
+        boolean isFromSender = false;
+        if (messageBus != null) {
+            isFromSender = messageBus.stream()
+                    .anyMatch(message -> message.isFromSender(senderId));
```
Comment (mprtcz) [inline src/main/java/com/spanish_inquisition/battleship/server/MessageBus.java]: 😮 👍

File: src/test/java/com/spanish_inquisition/battleship/common/AdjacentFieldsCalcTest.java [hunk 4]
```diff
@@ -4,9 +4,11 @@
 import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
+import static org.testng.Assert.*;
+
 public class AdjacentFieldsCalcTest {
 
-    @DataProvider(name = "IndexesProvider")
+    @DataProvider
```
Comment (mprtcz) [inline src/test/java/com/spanish_inquisition/battleship/common/AdjacentFieldsCalcTest.java]: Much clearer now 👍

File: src/test/java/com/spanish_inquisition/battleship/common/AdjacentFieldsCalcTest.java [hunk 5]
```diff
@@ -38,22 +40,22 @@
...
 
-    @Test(dataProvider = "CoordinatesProvider")
+    @Test(dataProvider = "coordinatesProvider")
     public void shouldTranslateCoordinatesToIndex(int[] coordinates, int expectedIndex){
         // When
         int calculatedIndex = AdjacentFieldsCalc.translateCoordinatesToIndex(coordinates[0],coordinates[1]);
 
         // Then
-        Assert.assertEquals(calculatedIndex, expectedIndex);
+        assertEquals(calculatedIndex, expectedIndex);
```
Comment (mprtcz) [inline src/test/java/com/spanish_inquisition/battleship/common/AdjacentFieldsCalcTest.java]: 👍 for static import

File: src/test/java/com/spanish_inquisition/battleship/server/MessageTest.java [hunk 6]
```diff
@@ -1,9 +1,10 @@
...
 
-import org.testng.Assert;
 import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
 
+import static org.testng.Assert.assertEquals;
+
 public class MessageTest {
 
     private final String content = "Test text";
```
Comment (Devas) [inline src/test/java/com/spanish_inquisition/battleship/server/MessageTest.java]: If it's final, i would use uppercase i.e. CONTENT

File: src/test/java/com/spanish_inquisition/battleship/server/MessageBusTest.java [hunk 7]
```diff
@@ -1,35 +1,111 @@
...
+                        20, true},
+                {Arrays.asList(
+                        new Message(1, 2, "AAA"),
+                        new Message(10, 20, "BBB"),
+                        new Message(1, 2, "CCC"),
+                        new Message(5, 36, "QWE")),
+                        77, false},
+        };
+    }
+    // endregion
```
Comment (mprtcz) [inline src/test/java/com/spanish_inquisition/battleship/server/MessageBusTest.java]: 🥇 for regions

File: src/test/java/com/spanish_inquisition/battleship/server/MessageBusTest.java [hunk 8]
```diff
@@ -38,9 +114,48 @@ public void shouldGetMessageForRecipient(List<Message> messages, int recipientId
...
+        boolean found = messageBus.haveMessageFromSender(senderId);
+
+        // Then
+        assertEquals(expected, found);
+    }
+
+    @Test(dataProvider = "checkingForRecipientMessagesProvider")
+    public void shouldCheckForMessageForRecipientInBus(List<Message> messages, int recipientId, boolean expected) {
+        // Given
+        messages.forEach(message -> messageBus.addMessage(
```
Comment (Devas) [inline src/test/java/com/spanish_inquisition/battleship/server/MessageBusTest.java]: :+1:

File: src/test/java/com/spanish_inquisition/battleship/server/MessageBusTest.java [hunk 9]
```diff
@@ -38,9 +114,48 @@ public void shouldGetMessageForRecipient(List<Message> messages, int recipientId
...
+        assertEquals(expectedMessage.getContent(), getMessage.getContent());
+        assertEquals(expectedMessage.getSender(), getMessage.getSender());
+        assertEquals(expectedMessage.getRecipient(), getMessage.getRecipient());
+        assertFalse(messageBus.haveMessageForRecipient(recipientId));
+    }
+
+    @Test(dataProvider = "gettingFromSenderMessageProvider")
+    public void shouldGetMessageForSender(List<Message> messages, int senderId, Message expectedMessage) {
+        // Given
+        messages.forEach(message -> messageBus.addMessage(
```
Comment (mprtcz) [inline src/test/java/com/spanish_inquisition/battleship/server/MessageBusTest.java]: Lambdas all over the place 💯

File: src/main/java/com/spanish_inquisition/battleship/server/MessageBus.java [hunk 10]
```diff
@@ -32,36 +31,37 @@ Message getMessageFor(int recipientId) {
...
-        if (messageBus != null) {
-            for (Message message : messageBus)
-                if (message.getSender() == senderId) return true;
+    Message getMessageFrom(int senderId) {
+        Message firstSenderMessage;
+        try {
+            firstSenderMessage = messageBus.stream()
+                    .filter(message -> message.isFromSender(senderId)).findFirst().get();
+        } catch (NoSuchElementException e) {
+            firstSenderMessage = new Message();
```
Comment (Devas) [inline src/main/java/com/spanish_inquisition/battleship/server/MessageBus.java]: Catch cannot execute logic

File: src/main/java/com/spanish_inquisition/battleship/server/Message.java [hunk 11]
```diff
@@ -5,6 +5,9 @@
     private int recipient;
     private String content;
 
+    public Message() {
```
Comment (Devas) [inline src/main/java/com/spanish_inquisition/battleship/server/Message.java]: Do you really need this empy constructor?

File: src/main/java/com/spanish_inquisition/battleship/server/BattleshipGame.java [hunk 12]
```diff
@@ -22,7 +22,7 @@ public void proceed() {
                 System.out.println("checking message from player " + player.getPlayerName());
                 int playerId = player.getClientId();
                 if(requestBus.haveMessageFromSender(playerId)){
-                    String message = requestBus.getMessageFromSender(playerId).getContent();
+                    String message = requestBus.getMessageFrom(playerId).getContent();
                     if(message.contains(Header.MOVE_REGULAR.name())) {
                         logger.log(DEFAULT_LEVEL, "Player clicked on field:" + message.substring(message.indexOf(":") + 1, message.length()));
```
Comment (Devas) [inline src/main/java/com/spanish_inquisition/battleship/server/BattleshipGame.java]: Magic number 1

**Pull request merged**